spring-cloud-gcp-secretmanager/src/main/java/org/springframework/cloud/gcp/secretmanager/SecretManagerOperations.java

```diff
@@ -111,4 +177,14 @@
 	 * @return true if the secret exists in Secret Manager; false otherwise
 	 */
 	boolean secretExists(String secretId);
+
+	/**
+	 * Returns true if there already exists a secret under the GCP {@code projectId} with the
+	 * {@code secretId}.
+	 *
+	 * @param secretId unique identifier of your secret in Secret Manager.
+	 * @param projectId unique identifier of your project.
+	 * @return true if the secret exists in Secret Manager; false otherwise
+	 */
+	boolean secretExists(String secretId, String projectId);
```

Comment:
Does it make sense to have `secretExists(String secretId, String versionName, String projectId)`?
This would check if a specific version of the secret exists.

Reply:
I guess that the primary use of this function is internal for the secret creation (create a new secret or add a version to an existing secret). Thereby, the version is useless in this context.

However, if we want to offer a convenient interface to developer, it could be interesting. If you agree, I propose these interfaces:

```
boolean secretExists(String secretId);// Check latest version in default project
boolean secretExists(String secretId, String version); // Check specific version in default project
boolean secretExists(String secretId, String version, String projectId);// // Check specific version in specific project
```
Like for your `getSecretString` comment, I won't be able to keep `boolean secretExists(String secretId, String projectId);` because of method signature conflict with `boolean secretExists(String secretId, String version);`

Let me know.

Reply:
Yeah let's omit adding `secretExists(String secretId, String versionName, String projectId)`, since it would conflict with existing method signature and I think its value is relatively limited as you say with it being primarily internal.

Reply:
Sorry for the noise, you are absolutely right about the method signature collisions. I didn't think of that.